### PR TITLE
Project 1. Network simulator. Resolve issues 9-12

### PR DESCRIPTION
**Pull request by AntoxaBarin**
Issues resolved in this PR: #9, #10, #11, #12.

- Add implementation of classes `Flow`, `NetworkSimulator`, `Packet`
- Add tests (`test/NetworkSimulatorTest.cpp`)

### CODE REVIEW

File: include/NetworkSimulator.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * Size of a packet in bytes.
+     */
+    static const std::uint32_t packet_size_bytes = 1024;
+
+    /**
+     * Size of a receiver's bandwidth in bytes.
+     */
+    static const std::uint64_t bandwidth_bytes = 6'250'000'000;
+private:
+    Flow* flow;
```
Comment (IDKWNTCMF) [inline include/NetworkSimulator.hpp:32]: I suggest using smart pointers instead of raw ones

File: include/Server.hpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+protected:
+    std::uint32_t id;
+    std::uint64_t current_time_ns = 0;
+};
+
+// Type for sending server, contains ID and distance to receiver
+struct ServerSender : ServerBase {
+    ServerSender(std::uint64_t id, std::uint32_t distance_ns);
+    ServerSender() = default;
+    ~ServerSender() = default;
```
Comment (drgulevich) [inline include/Server.hpp:51]:
```suggestion
    virtual ~ServerSender() = default;
```

File: src/Flow.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Flow::Flow(const std::vector<std::uint32_t>& distances) {
+    destination_node = ServerBase(0);
+    for (std::uint32_t distance : distances) {
+        source_nodes.emplace_back(++last_given_server_id, distance);
+    }
+}
+
+void Flow::Send() {
+    for (auto& s: source_nodes) {
```
Comment (drgulevich) [inline src/Flow.cpp:14]:
```suggestion
    for (ServerSender& s: source_nodes) {
```

File: src/Flow.cpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        for (std::uint64_t i = 0; i < s.GetCWNDSize(); ++i) {
+            packets.emplace(s.GetID(), s.GetCurrentTime(), s.GetCurrentTime() + s.GetDistance());
+            s.IncreaseCurrentTime(s.GetDistance());
+        }
+    }
+}
+
+void Flow::PrintQueueSize() const {
+    std::cout << "Priority queue size: " << packets.size() << ".\n";
```
Comment (drgulevich) [inline src/Flow.cpp:23]: Maybe make this method as a getter `GetQueueSize()` and use it in `std::cout` outide the class

File: src/Flow.cpp (hunk 3 above)
Comment (drgulevich) [inline src/Flow.cpp:14]: `    for (ServerSender& s: source_nodes) {`

File: include/Server.hpp (hunk 2 above)
Comment (drgulevich) [inline include/Server.hpp:51]: `virtual ~ServerSender() = default;`

**Comment by AntoxaBarin** [comment]
Thank for review! I will close this PR and create separated PR's for each issue (#9, #10, #11, #12).

**Pull request closed without merging**